src/PostgREST/Auth.hs

```diff
@@ -163,6 +163,8 @@ middleware appState app req respond = do
       jwtCacheState = getJwtCacheState appState
 
 -- If ServerTimingEnabled -> calculate JWT validation time
+      -- Now that we also calculate the timings when Prefer: metrics=timings,
+      -- how do we get that preference here?
```

Comment:
@steve-chavez We need the preference here too. I am not sure about this but I think we should think about storing the `ApiRequest` result in `AppState`. This way it would be easier to access already parsed request information at any stage in the code.

Reply:
I think we should just refactor to remove the `Auth.middleware` approach. I've never liked that one as it hides side effects and obscures logic. It might allow us to remove the `unsafePerformIO` from Auth.hs too:

https://github.com/PostgREST/postgrest/blob/e4a2095b5a997e721299c24280ab785317b50fa0/src/PostgREST/Auth.hs#L190-L198

Once that's done you should be able to use/get the prefer value value before validating the JWT.

Reply:
Agreed. Let me refactor this.